diff -Naur a/bridge/fdb.c b/bridge/fdb.c
--- a/bridge/fdb.c	2020-03-28 01:48:35.695836272 -0300
+++ b/bridge/fdb.c	2020-03-28 01:50:16.975841274 -0300
@@ -18,7 +18,6 @@
 #include <net/if.h>
 #include <netinet/in.h>
 #include <linux/if_bridge.h>
-#include <linux/if_ether.h>
 #include <linux/neighbour.h>
 #include <string.h>
 #include <limits.h>
diff -Naur a/bridge/mdb.c b/bridge/mdb.c
--- a/bridge/mdb.c	2020-03-28 01:48:35.695836272 -0300
+++ b/bridge/mdb.c	2020-03-28 01:50:16.975841274 -0300
@@ -11,7 +11,6 @@
 #include <net/if.h>
 #include <netinet/in.h>
 #include <linux/if_bridge.h>
-#include <linux/if_ether.h>
 #include <string.h>
 #include <arpa/inet.h>
 
diff -Naur a/bridge/vlan.c b/bridge/vlan.c
--- a/bridge/vlan.c	2020-03-28 01:48:35.695836272 -0300
+++ b/bridge/vlan.c	2020-03-28 01:50:16.975841274 -0300
@@ -7,7 +7,6 @@
 #include <net/if.h>
 #include <netinet/in.h>
 #include <linux/if_bridge.h>
-#include <linux/if_ether.h>
 #include <string.h>
 
 #include "json_print.h"
diff -Naur a/include/uapi/linux/if_bridge.h b/include/uapi/linux/if_bridge.h
--- a/include/uapi/linux/if_bridge.h	2020-03-28 01:48:35.725836274 -0300
+++ b/include/uapi/linux/if_bridge.h	2020-03-28 01:50:16.975841274 -0300
@@ -14,8 +14,9 @@
 #ifndef _LINUX_IF_BRIDGE_H
 #define _LINUX_IF_BRIDGE_H
 
+#include <netinet/if_ether.h>
+
 #include <linux/types.h>
-#include <linux/if_ether.h>
 #include <linux/in6.h>
 
 #define SYSFS_BRIDGE_ATTR	"bridge"
diff -Naur a/include/uapi/linux/if_tunnel.h b/include/uapi/linux/if_tunnel.h
--- a/include/uapi/linux/if_tunnel.h	2020-03-28 01:48:35.725836274 -0300
+++ b/include/uapi/linux/if_tunnel.h	2020-03-28 01:50:16.975841274 -0300
@@ -2,8 +2,9 @@
 #ifndef _IF_TUNNEL_H_
 #define _IF_TUNNEL_H_
 
+#include <net/if.h>
+
 #include <linux/types.h>
-#include <linux/if.h>
 #include <linux/ip.h>
 #include <linux/in6.h>
 #include <asm/byteorder.h>
diff -Naur a/include/uapi/linux/ip6_tunnel.h b/include/uapi/linux/ip6_tunnel.h
--- a/include/uapi/linux/ip6_tunnel.h	2020-03-28 01:48:35.725836274 -0300
+++ b/include/uapi/linux/ip6_tunnel.h	2020-03-28 01:50:16.975841274 -0300
@@ -2,8 +2,9 @@
 #ifndef _IP6_TUNNEL_H
 #define _IP6_TUNNEL_H
 
+#include <net/if.h>		/* For IFNAMSIZ. */
+
 #include <linux/types.h>
-#include <linux/if.h>		/* For IFNAMSIZ. */
 #include <linux/in6.h>		/* For struct in6_addr. */
 
 #define IPV6_TLV_TNL_ENCAP_LIMIT 4
diff -Naur a/include/uapi/linux/netdevice.h b/include/uapi/linux/netdevice.h
--- a/include/uapi/linux/netdevice.h	2020-03-28 01:48:35.725836274 -0300
+++ b/include/uapi/linux/netdevice.h	2020-03-28 01:50:16.975841274 -0300
@@ -26,7 +26,8 @@
 #ifndef _LINUX_NETDEVICE_H
 #define _LINUX_NETDEVICE_H
 
-#include <linux/if.h>
+#include <net/if.h>
+
 #include <linux/if_ether.h>
 #include <linux/if_packet.h>
 #include <linux/if_link.h>
diff -Naur a/ip/ip6tunnel.c b/ip/ip6tunnel.c
--- a/ip/ip6tunnel.c	2020-03-28 01:48:35.745836275 -0300
+++ b/ip/ip6tunnel.c	2020-03-28 01:50:16.975841274 -0300
@@ -27,8 +27,8 @@
 #include <sys/socket.h>
 #include <arpa/inet.h>
 #include <sys/ioctl.h>
+#include <net/if.h>
 #include <linux/ip.h>
-#include <linux/if.h>
 #include <linux/if_arp.h>
 #include <linux/if_tunnel.h>
 #include <linux/ip6_tunnel.h>
diff -Naur a/ip/ipl2tp.c b/ip/ipl2tp.c
--- a/ip/ipl2tp.c	2020-03-28 01:48:35.745836275 -0300
+++ b/ip/ipl2tp.c	2020-03-28 01:50:16.975841274 -0300
@@ -19,7 +19,7 @@
 #include <sys/socket.h>
 #include <arpa/inet.h>
 #include <sys/ioctl.h>
-#include <linux/if.h>
+#include <net/if.h>
 #include <linux/if_arp.h>
 #include <linux/ip.h>
 
diff -Naur a/ip/iplink_hsr.c b/ip/iplink_hsr.c
--- a/ip/iplink_hsr.c	2020-03-28 01:48:35.745836275 -0300
+++ b/ip/iplink_hsr.c	2020-03-28 01:50:16.975841274 -0300
@@ -15,7 +15,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <sys/socket.h>  /* Needed by linux/if.h for some reason */
-#include <linux/if.h>
+#include <net/if.h>
 #include <linux/if_arp.h>
 #include "rt_names.h"
 #include "utils.h"
diff -Naur a/ip/ipmaddr.c b/ip/ipmaddr.c
--- a/ip/ipmaddr.c	2020-03-28 01:48:35.745836275 -0300
+++ b/ip/ipmaddr.c	2020-03-28 01:50:16.975841274 -0300
@@ -19,9 +19,9 @@
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <string.h>
+#include <net/if.h>
 
 #include <linux/netdevice.h>
-#include <linux/if.h>
 #include <linux/if_arp.h>
 #include <linux/sockios.h>
 
diff -Naur a/ip/ipmroute.c b/ip/ipmroute.c
--- a/ip/ipmroute.c	2020-03-28 01:48:35.745836275 -0300
+++ b/ip/ipmroute.c	2020-03-28 01:50:16.975841274 -0300
@@ -20,9 +20,9 @@
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <string.h>
+#include <net/if.h>
 
 #include <linux/netdevice.h>
-#include <linux/if.h>
 #include <linux/if_arp.h>
 #include <linux/sockios.h>
 
diff -Naur a/ip/iptuntap.c b/ip/iptuntap.c
--- a/ip/iptuntap.c	2020-03-28 01:48:35.745836275 -0300
+++ b/ip/iptuntap.c	2020-03-28 01:50:16.985841275 -0300
@@ -18,7 +18,6 @@
 #include <sys/socket.h>
 #include <arpa/inet.h>
 #include <sys/ioctl.h>
-#include <linux/if.h>
 #include <linux/if_tun.h>
 #include <linux/if_arp.h>
 #include <pwd.h>
diff -Naur a/ip/tunnel.c b/ip/tunnel.c
--- a/ip/tunnel.c	2020-03-28 01:48:35.755836275 -0300
+++ b/ip/tunnel.c	2020-03-28 01:50:16.985841275 -0300
@@ -30,7 +30,7 @@
 #include <sys/socket.h>
 #include <sys/ioctl.h>
 #include <netinet/in.h>
-#include <linux/if.h>
+#include <net/if.h>
 #include <linux/ip.h>
 #include <linux/if_tunnel.h>
 #include <linux/if_arp.h>
