diff -Naur a/Makefile b/Makefile
--- a/Makefile	2019-10-07 11:55:39.000000000 -0300
+++ b/Makefile	2019-11-02 04:11:06.019531117 -0300
@@ -41,8 +41,6 @@
 	chvt              \
 	clear             \
 	ctrlaltdel        \
-	dd                \
-	df                \
 	dmesg             \
 	eject             \
 	fallocate         \
@@ -52,7 +50,6 @@
 	getty             \
 	halt              \
 	hwclock           \
-	id                \
 	insmod            \
 	killall5          \
 	last              \
@@ -60,8 +57,6 @@
 	login             \
 	lsmod             \
 	lsusb             \
-	mesg              \
-	mknod             \
 	mkswap            \
 	mount             \
 	mountpoint        \
@@ -70,7 +65,6 @@
 	passwd            \
 	pidof             \
 	pivot_root        \
-	ps                \
 	pwdx              \
 	readahead         \
 	respawn           \
@@ -87,27 +81,20 @@
 	unshare           \
 	uptime            \
 	vtallow           \
-	watch             \
-	who
+	watch
 
 MAN1 = \
 	chvt.1              \
 	clear.1             \
-	dd.1                \
-	df.1                \
 	dmesg.1             \
 	eject.1             \
 	fallocate.1         \
 	free.1              \
-	id.1                \
 	login.1             \
-	mesg.1              \
-	mknod.1             \
 	mountpoint.1        \
 	pagesize.1          \
 	passwd.1            \
 	pidof.1             \
-	ps.1                \
 	pwdx.1              \
 	respawn.1           \
 	stat.1              \
@@ -116,8 +103,7 @@
 	unshare.1           \
 	uptime.1            \
 	vtallow.1           \
-	watch.1             \
-	who.1
+	watch.1
 
 MAN8 = \
 	ctrlaltdel.8        \
